ext/standard/charrepr.h

```diff
@@ -0,0 +1,284 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Yihang Wang <wangyihanger@gmail.com>                         |
+  +----------------------------------------------------------------------+
+*/
+
+#ifndef STRING_H
+#define STRING_H
+
+typedef struct {
+    const char *repr; 
+    const unsigned char len;
+} char_repr_t;
+
+static const char_repr_t char_reprs[256] = {
+	{"\\x00", 4},
+	{"\\x01", 4},
+	{"\\x02", 4},
+	{"\\x03", 4},
+	{"\\x04", 4},
+	{"\\x05", 4},
+	{"\\x06", 4},
+	{"\\x07", 4},
+	{"\\b", 2},
+	{"\\t", 2},
+	{"\\n", 2},
+	{"\\x0b", 4},
+	{"\\f", 2},
+	{"\\r", 2},
+	{"\\x0e", 4},
+	{"\\x0f", 4},
+	{"\\x10", 4},
+	{"\\x11", 4},
+	{"\\x12", 4},
+	{"\\x13", 4},
+	{"\\x14", 4},
+	{"\\x15", 4},
+	{"\\x16", 4},
+	{"\\x17", 4},
+	{"\\x18", 4},
+	{"\\x19", 4},
+	{"\\x1a", 4},
+	{"\\x1b", 4},
+	{"\\x1c", 4},
+	{"\\x1d", 4},
+	{"\\x1e", 4},
+	{"\\x1f", 4},
+	{" ", 1},
+	{"!", 1},
+	{"\\\"", 2},
+	{"#", 1},
+	{"$", 1},
+	{"%%", 1},
```

Comment:
This looks fishy. `%%` only needs to be escaped in `printf`. Given the impl uses `memcpy` rather than `strcpy`, it shouldn't lead to a bug though.

Reply:
> Was this code copied from somewhere? There's no header/license that would indicate so.

Actually, I just generated this header file by a few lines of ad-hoc Python code.

Reply:
> This looks fishy. %% only needs to be escaped in printf. Given the impl uses memcpy rather than strcpy, it shouldn't lead to a bug though.

Ahhhh, yes, I will fix it soon.

Reply:
@iluuu1994 Fixed in 32340baf165f7ec8d0391fa6bdcb37eca3791403.

https://github.com/php/php-src/pull/15730/commits/32340baf165f7ec8d0391fa6bdcb37eca3791403#diff-9dc13ac25ae79e9c7bd257341e953577aae1b2cb7562fb9acd64ce06c1baa71eL63-R63